core/src/main/java/nl/nn/adapterframework/extensions/esb/EsbJmsListener.java

```diff
@@ -102,21 +103,21 @@ private void configurexPathLogging() {
 	}
 
 	@Override
-	protected String retrieveIdFromMessage(Message message, Map<String, Object> threadContext) throws ListenerException {
-		String id = super.retrieveIdFromMessage(message, threadContext);
+	public Map<String, Object> populateContextFromMessage(Message rawMessage) throws ListenerException {
+		Map<String, Object> messageContext = super.populateContextFromMessage(rawMessage);
 		if (isCopyAEProperties()) {
 			Enumeration<?> propertyNames = null;
 			try {
-				propertyNames = message.getPropertyNames();
+				propertyNames = rawMessage.getPropertyNames();
 			} catch (JMSException e) {
 				log.debug("ignoring JMSException in getPropertyName()", e);
 			}
 			while (propertyNames.hasMoreElements()) {
 				String propertyName = (String) propertyNames.nextElement ();
 				if (propertyName.startsWith("ae_")) {
 					try {
-						Object object = message.getObjectProperty(propertyName);
-						threadContext.put(propertyName, object);
+						Object object = rawMessage.getObjectProperty(propertyName);
+						messageContext.put(propertyName, object);
```

Comment:
Voor de backwards compatibility is dit niet zo handig, maar deze methode is ook gedeprecate dacht ik?
De JMSFacade bewaard nu ook al alle headers, dus als je dit wilt, kan het er beter uit.

Reply:
Ik doe in feite niets anders dan al werd gedaan alleen werd alles gezet in een variabele met andere naam... Die naam was inderdaad niet zo goed en uiteindelijk blijkt het de PipeLineSession in te moeten, dus ik probeer nu dingen een wat betere naam te geven, en pas elders een parameter-type aan naar PipeLineSession.

(De naam threadContext was al fout, als ik nu door de dingen heenloop).